### ISSUE TITLE
Harness Meta-Data standardization

### ISSUE THREAD

**Issue by exodist** [Test-More/Test2-Harness#77]
Right now yath allows harness specific meta-data to be noted in test headers:
```
# HARNESS-NO-PRELOAD
# HARNESS-NO-STREAM
# HARNESS-CATEGORY-FOO
```

These are very useful for controlling harness behavior for tests that are special snowflakes in one way or another. Having the meta-data be directly inside the test file is also really handy, and easy to spot/modify.

I am however beginning to think the current system yath uses is hard to parse, and open to a lot of inconsistency. Things are being added fairly organically.

I think we need to write a standard for harness directives / meta-data. Ideally this will be fairly language agnostic. Maybe something as simple as a json blob with an easy to find, unambiguous label. This blob could easily be added inside a quoted string, multi-line comment, multiple single-line comments with an identical prefix (ie # or // depending on language).

```
=pod OR qw| OR /*

[ANYTHING] test-harness-meta {
  features: {
    preload: false,
    stream: true,
  },
  category: 'foo',
  ...
}

=cut OR | or */
```

We would have some standard keys (features, category, etc). and a 'custom' key for harness/language/codebase specific additional data.

Please discuss.

**Comment by autarch** [comment]
I agree that making this more of a proper data structure is a good idea. I think you want a `=begin Test2-Harness` as the marker. That's the standard way in pod to include arbitrary chunks of data.

**Comment by nawglan** [comment]
allowing comments on lines inside the block would be good too if possible.

**Comment by exodist** [comment]
@nawglan can you put comments into json blobs and have parsers like JSON-XS work? I am not opposed to some pre-processing (needed if structure is added with single-line comment prefixes like # and //)

**Comment by nawglan** [comment]
I was thinking of stripping the single line comments out before passing to the parser.   Some json parsers allow a "relaxed" mode which ignores comments.

**Comment by atoomic** [comment]
I understand the will for a proper/better format, I'm also fine with the existing syntax, which as @toddr pointed allows comments. And has the advantage to easily show in 'git grep' for example.

Most of our tests use no comments at all, and the few ones that need some tweak use one or two of directive maximum... 

Using a simple line tag, make it easy to grep/tag/edit

I could see moving to a different format over time, but this would make more sense once we have more features and start adding many (~>4) directives to tests.

**Comment by toddr** [comment]
If I were going to amend the existing policy in any way, I might suggest that the the comments have to be before any code, so we'd stop pasting on the first non-blank or non-comment line.
